core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -866,7 +866,7 @@ class LogTest {
 
     log.highWatermark = log.logEndOffset
     log.maybeIncrementLogStartOffset(1L)
-    log.onHighWatermarkIncremented(log.logEndOffset)
+    log.highWatermarkMetadata = LogOffsetMetadata(log.logEndOffset)
```

Comment:
nit: could probably use `log.highWatermark = {}` for all of these

Reply:
Done. Missed this comment. Going to push one commit with fix on top of current ones.